### ISSUE TITLE
`help(boost_histogram.histogram)` produces unhelpful result

### ISSUE THREAD

**Issue by HDembinski** [scikit-hep/boost-histogram#174]
```
Help on FactoryMeta in boost_histogram object:

boost_histogram.histogram = class FactoryMeta(builtins.object)
 |  boost_histogram.histogram(f, types)
 |
 |  Methods defined here:
 |
 |  __call__(self, *args, **kwargs)
 |      Call self as a function.
 |
 |  __init__(self, f, types)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  __instancecheck__(self, other)
 |      Check if an object is an instance.
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors defined here:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
```
The expectation is to get help on the histogram object.

**Comment by henryiii** [comment]
I think this will be fixed by the redesign for #155, but if not, it should fixed.

**Comment by henryiii** [comment]
PS: Docstrings are in sore need of attention throughout the library, actually, and should be addressed when we work on the docs.
